src/book/02-projects/01-nubots/12-mathematics.mdx

```diff
@@ -0,0 +1,250 @@
+
+We currently use the [Eigen](https://eigen.tuxfamily.org/) C++ template library for linear algebra in the NUbots codebase.
+
+## Bases
+
+A **basis** is a **maximal** set of **linearly independent** vectors.
+
+In this context, bases usually span $\mathbb{R}^3$. That is, they cover **three dimensional space**. Given a basis with three vectors $\mathbf{b}_1, \mathbf{b}_2, \mathbf{b}_3$, Any three dimensional vector $\mathbf{u}$ can be written as
+
+$$
+\mathbf{u} = a_1 \cdot \mathbf{b}_1 + a_2 \cdot \mathbf{b}_2 + a_3 \cdot \mathbf{b}_3
+$$
+
+where $a_1, a_2, a_3 \in \mathbb{R}$. In other words, $a_1, a_2, a_3$ are scalars. These scalars are **unique**.
+
+If the vectors in the basis are mutually perpendicular, then they are **orthogonal**.
+
+An **orthonormal basis (ONB)** is a basis which is orthogonal and each vector has length 1.
+
+An example of an OBN is the **standard basis**
+
+$$
+e_1 = (1, 0, 0) \quad
+e_2 = (0, 1, 0) \quad
+e_3 = (0, 0, 1)
+$$
+
+## Vectors
+
+As mentioned in the previous section, any vector $\mathbf{r}$ can be expressed as a linear combination of the elements of a basis
+
+$$
+\mathbf{r}^a = r_1^a \mathbf{a}_1 + r_2^a \mathbf{a}_2 + r_3^a \mathbf{a}_3
+$$
+
+The scalars $r_1^a, r_2^a, r_3^a$ are the **coordinates** for the vector $\mathbf{r}$ in the basis $a$.
+
+In the NUbots codebase, vectors are written as `rABb` which denotes a vector from point $B$ to point $A$ in the basis (or **space**) $b$. These are column matrices of the coordinates of the vectors
+
+$$
+\mathbf{r}^a = \begin{bmatrix}
+r_1^a \\
+r_2^a \\
+r_3^a \end{bmatrix}
+$$
+
+To make computations such as **inner product** or **cross product** with coordinates, the vectors must be relative to the **same basis**.
+
+When drawing a vector $\mathbf{r}_{AB}^b$, that is a vector from $B$ to $A$ in the space $b$, the **tail** is at $B$ and the **head** is at $A$. **Vector addition** requires that the head of one vector is the same as the tail of the other. This is demonstrated [here](http://mathworld.wolfram.com/VectorAddition.html). We have that
+
+$$
+\mathbf{r}^b_{AC} = \mathbf{r}^b_{AB} + \mathbf{r}^b_{BC} = \mathbf{r}^b_{BC} + \mathbf{r}^b_{AB}
+$$
+
+since vector addition is **commutative**. This is written in code as
+
+```cpp
+Eigen::Vector3d rACb = rABb + rBCb;
+```
+
+## Rotations
+
+To describe the motion of the robot, we need a **reference frame** and a **coordinate system**.
+
+The reference frame is the **physical rigid body** and the coordinate system is a **mathematical concept**, which is represented as a **basis**.
+
+NUbots use a **right-handed coordinate system** where the x-axis is **forward**, the y-axis is to the **left** and the z-axis is **up**.
```

Comment:
Probably too much to ask but a diagram/illustration with hand would be great here.

Reply:
Something like the one on Wikipedia: https://en.wikipedia.org/wiki/Right-hand_rule